carAPI/src/main/java/com/everis/boundary/CarResource.java

```diff
@@ -63,31 +52,31 @@ public Response getCar(@PathParam("carId") long id) {
 	//Recibe un coche y lo a�ade al array
 	@POST
 	public Response createCar(Car car, @Context UriInfo uriInfo) {
-		cars.put(car.getId(), car);
-		String newId = String.valueOf(car.getId());
+		Car newCar = carService.createCar(car);
+		String newId = String.valueOf(newCar.getId());
 		URI uri = uriInfo.getAbsolutePathBuilder().path(newId).build();
-		return Response.created(uri).entity(car).build();
+		return Response.created(uri).entity(newCar).build();
 	}
 
 	//Recibe un coche y actualiza sus datos
 	@PUT
 	@Path("/{carId}")
 	public Response updateCar(@PathParam("carId") long id, Car car) {
 		car.setId(id);
-		if(car.getId()<=0) {
-			//Devolver mediante objeto error
+		Car newCar = carService.updateCar(car);
+		if(newCar==null) {
```

Comment:
The method in the service layer return 'null' when the id is a negative number, therefore you're returning NOT_FOUND only in case of negative id, but what if the id really does not exist ?